### PR TITLE
Added IRLML6401

### PR DESCRIPTION
**Pull request by marecl**
Added IRLML6401 P-MOSFET Hexfet (with parameters and datasheet) to Transistor_FET library
https://www.infineon.com/dgdl/irlml6401pbf.pdf?fileId=5546d462533600a401535668b96d2634

Corrected IRLML6402 description to HEXFET Power MOSFET (like other Infineon parts are described)

Can't provide screenshot as it's added together with IRLML6402 as alias of TP0610T
Uses standard SOT-23 package
Pinout matches with IRLML6401 and TP0610T

---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the symbol(s) you are contributing
- [ ] Provide a screenshot of the symbol(s) from the symbol editor with the pin types visible
- [x] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
  - A new fitting footprint must be submitted if the library does not yet contain one.
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [x] Check the output of the Travis automated check scripts - fix any errors as required
- [x] Give a reason behind any intentional library convention rule violation.

---

Be patient, we maintainers are volunteers with limited time and need to check your contribution against the datasheet. You can speed up the process by providing all the necessary information (see above). And you can speed up the process even more by providing additional info like the screenshot of the symbol editor pin table (or for high pin counts converted to csv) sorted in the same way as the pin table in the datasheet and a direct link to the datasheet page that contains the pin table.

### CODE REVIEW

**Comment by aris-kimi** [comment]
Hi @marecl , thank you for this. If i may suggest, keep description as other aliases, and place `HEXFET®` to the keyword area. I also think it would be better if `, 65mOhm Rds` is placed before `, P-Channel MOSFET` in the description.
Thank you for your time, it was just a suggestion. you better wait for a review..

**Comment by marecl** [comment]
@aris-kimi Thanks for the suggestion. I used IRLML6402 as a template for the new part, but your idea sounds better.
I have some time so I'll adjust other names and add keywords where necessary.

**Comment by cpresser** [comment]
Hi @marecl, Thanks for you first contribution.
Usually we don't accept pull-requests that add parts and change parts at the same time. I will make an exception and review as it is, you had good intentions and that should be honored.
Next time, please split additions and enhancements :smile:

**Comment by marecl** [comment]
hi @cpresser thank you very much! I will remember next time :)

File: Transistor_FET.dcm [hunk 1]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D -3.6A Id, -30V Vds, 64mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
 F https://www.infineon.com/dgdl/irlml9301pbf.pdf?fileId=5546d462533600a401535668e5e42640
 $ENDCMP
 #
 $CMP IRLZ34N
-D 30A Id, 55V Vds, Single N-Channel HEXFET Power MOSFET, 35mOhm Ron, TO-220AB
-K Single N-Channel HEXFET Power MOSFET
+D 30A Id, 55V Vds, 35mOhm Rds, N-Channel HEXFET Power MOSFET, TO-220AB
+K N-Channel HEXFET MOSFET Logic-Level
```
Comment (cpresser) [inline Transistor_FET.dcm:2519]: Same as above, I could not find the word `logic in the datasheet.

File: Transistor_FET.dcm [hunk 2]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D -3.7A Id, -20V Vds, 65mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
 F https://www.infineon.com/dgdl/irlml6402pbf.pdf?fileId=5546d462533600a401535668d5c2263c
 $ENDCMP
 #
 $CMP IRLML9301
-D -3.6A Id, -30V Vds, P-Channel HEXFET Power MOSFET, 64mOhm Rds, SOT-23
-K P-Channel MOSFET
+D -3.6A Id, -30V Vds, 64mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
```
Comment (cpresser) [inline Transistor_FET.dcm:2513]: Same as above, I could not find the word `logic in the datasheet.

File: Transistor_FET.dcm [hunk 3]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D -4.3A Id, -12V Vds, 50mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
+F https://www.infineon.com/dgdl/irlml6402pbf.pdf?fileId=5546d462533600a401535668d5c2263c
+$ENDCMP
+#
 $CMP IRLML6402
-D -3.7A Id, -20V Vds, P-Channel MOSFET, 65mOhm Rds, SOT-23
-K P-Channel MOSFET
+D -3.7A Id, -20V Vds, 65mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
```
Comment (cpresser) [inline Transistor_FET.dcm:2507]: Same as above, I could not find the word `logic in the datasheet.

**Review by cpresser — CHANGES_REQUESTED** [review]
I am not sure about all the `logic-level` keywords.
Most of them can be used with 3.3V logic-levels. But that depends on the application. For some of them 5V Ugs seems like a better value to properly drive them.
We can have a lengthy discussion what would be appropriate here. But I would argue to just go with the datasheet - even it its from 1997 (and thinks 5V is logic-level).

File: Transistor_FET.dcm [hunk 4]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
 $ENDCMP
 #
 $CMP IRF9394M
-D -14A Id, -30V Vds, 6.5mOhm Rds, P-Channel MOSFET, DirectFET MC
-K P-Channel MOSFET
+D -14A Id, -30V Vds, 6.5mOhm Rds, Dual P-Channel MOSFET, DirectFET MC
+K Dual P-Channel MOSFET HEXFET
```
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2381]: I'd assume the symbol is incorrect then. It has common drain but separate gates and sources making total of 4 pins for the footprint.
image

File: Transistor_FET.dcm (hunk 1 above)
Comment (marecl) [reply Transistor_FET.dcm:2519]: Logic Level gate drive
image

File: Transistor_FET.dcm (hunk 2 above)
Comment (marecl) [reply Transistor_FET.dcm:2513]: Might be true. Product webpage claims it's "logic level optimized" for 4.5 and 2.5V
https://www.infineon.com/cms/en/product/power/mosfet/12v-300v-n-channel-power-mosfet/irlml9301/

File: Transistor_FET.dcm (hunk 3 above)
Comment (marecl) [reply Transistor_FET.dcm:2507]: You've got me on this one

File: Transistor_FET.dcm [hunk 5]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
-K P-Channel MOSFET
+D -3.0A Id, -30V Vds, 98mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
 F https://www.infineon.com/dgdl/irlml5203pbf.pdf?fileId=5546d462533600a40153566868da261d
 $ENDCMP
 #
+$CMP IRLML6401
+D -4.3A Id, -12V Vds, 50mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
+F https://www.infineon.com/dgdl/irlml6402pbf.pdf?fileId=5546d462533600a401535668d5c2263c
```
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2502]: It doesn't but "1.8V gate rated" gives logic-level feel.
image

File: Transistor_FET.dcm [hunk 6]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D 1.2A Id, 60V Vds, 480mOhm Rds, N-Channel HEXFET Power MOSFET, SOT-23
+K N-Channel HEXFET MOSFET Logic-Level
 F https://www.infineon.com/dgdl/irlml2060pbf.pdf?fileId=5546d462533600a401535664b7fb25ee
 $ENDCMP
 #
 $CMP IRLML5203
-D -3.0A Id, -30V Vds, P-Channel HEXFET Power MOSFET, 98mOhm Rds, SOT-23
-K P-Channel MOSFET
+D -3.0A Id, -30V Vds, 98mOhm Rds, P-Channel HEXFET Power MOSFET, SOT-23
+K P-Channel HEXFET MOSFET Logic-Level
```
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2495]: Corrected

File: Transistor_FET.dcm [hunk 7]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D 5.3A Id, 30V Vds, 27mOhm Rds, N-Channel HEXFET Power MOSFET, SOT-23
+K N-Channel HEXFET MOSFET Logic-Level
 F https://www.infineon.com/dgdl/irlml0030pbf.pdf?fileId=5546d462533600a401535664773825df
 $ENDCMP
 #
 $CMP IRLML2060
-D 1.2A Id, 60V Vds, N-Channel MOSFET, 480mOhm Rds, SOT-23
-K N-Channel MOSFET
+D 1.2A Id, 60V Vds, 480mOhm Rds, N-Channel HEXFET Power MOSFET, SOT-23
+K N-Channel HEXFET MOSFET Logic-Level
```
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2489]: Corrected. Added the keyword by mistake after seeing "Normal-Level" in the desctiption.

File: Transistor_FET.dcm [hunk 8]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D 30A Id, 55V Vds, 22mOhm Rds, N-Channel HEXFET Power MOSFET, TO-220AB
+K N-Channel HEXFET MOSFET Logic-Level
 F http://www.irf.com/product-info/datasheets/data/irliz44n.pdf
 $ENDCMP
 #
 $CMP IRLML0030
-D 5.3A Id, 30V Vds, N-Channel HEXFET Power MOSFET, 27mOhm Rds, SOT-23
-K N-Channel HEXFET Power MOSFET
+D 5.3A Id, 30V Vds, 27mOhm Rds, N-Channel HEXFET Power MOSFET, SOT-23
+K N-Channel HEXFET MOSFET Logic-Level
```
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2483]: IRLML0030: optimized for 5V gate-drive voltage
https://www.infineon.com/cms/en/product/power/mosfet/12v-300v-n-channel-power-mosfet/irlml0030/

IRLIZ44N: Website described it as Logic-Level
https://www.infineon.com/cms/en/product/power/mosfet/12v-300v-n-channel-power-mosfet/irliz44n/

File: Transistor_FET.dcm [hunk 9]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D 375A Id, 40V Vds, 0.45mOhm Rds, N-Channel HEXFET Power MOSFET, DirectFET L8
+K N-Channel HEXFET MOSFET Logic-Level IRFET
 F https://www.infineon.com/dgdl/Infineon-IRL7472L1-DS-v02_00-EN.pdf?fileId=5546d46254e133b401555d17178250d8
 $ENDCMP
 #
 $CMP IRLB8721PBF
-D 62A Id, 30V Vds, N-Channel MOSFET, TO-220
-K N-Channel HEXFET Power MOSFET
+D 62A Id, 30V Vds, 8.7 mOhm Rds, N-Channel HEXFET Power MOSFET, TO-220
+K N-Channel HEXFET MOSFET Logic-Level
```
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2471]: IRL7472L1: Optimized for logic-level drive, but I think you're right as it's designed for normal-level
https://www.infineon.com/cms/en/product/power/mosfet/12v-300v-n-channel-power-mosfet/irl7472l1/

IRLB8721PBF: Optimized for Logic-Level gate drive (5V)
https://www.infineon.com/cms/en/product/power/mosfet/12v-300v-n-channel-power-mosfet/irlb8721/

File: Transistor_FET.dcm [hunk 10]
```diff
@@ -2377,146 +2377,152 @@ F https://www.infineon.com/dgdl/irf9383mpbf.pdf?fileId=5546d462533600a4015356116
...
+D 38A Id, 20V Vds, 0.75mOhm Rds, N-Channel HEXFET Power MOSFET, DirectFET MD
+K N-Channel HEXFET MOSFET IRFET
 F https://www.infineon.com/dgdl/irl6283mpbf.pdf?fileId=5546d462533600a40153565fe9452573
 $ENDCMP
 #
 $CMP IRL6297SD
-D 15A Id, 20V Vds, 4.9mOhm Rds, Dual N-Channel MOSFET, DirectFET SA
-K Dual N-Channel MOSFET
+D 15A Id, 20V Vds, 4.9mOhm Rds, Dual N-Channel HEXFET Power MOSFET, DirectFET SA
+K Dual N-Channel HEXFET MOSFET Logic-Level
```
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2459]: IRL6297SD: Super Logic-Level optimized for 4.5V gate drive voltage, capable of 2.5V (...)
https://www.infineon.com/cms/en/product/power/mosfet/20v-100v-n-channel-small-power-mosfet/irl6297sd/

File: Transistor_FET.dcm (hunk 4 above)
[the remark this answers is not in the document]
Comment (cpresser) [reply Transistor_FET.dcm:2393]: I agree. It is inconsistent. My proposal is to fix that bug in a dedicated PR. This one is already big enough.

**Review by cpresser — APPROVED** [review]

File: Transistor_FET.dcm (hunk 5 above)
[the remark this answers is not in the document]
Comment (cpresser) [reply Transistor_FET.dcm:2514]: The datasheet-link is wrong. It should be
https://www.infineon.com/dgdl/irlml6401pbf.pdf?fileId=5546d462533600a401535668b96d2634

File: Transistor_FET.dcm (hunk 1 above)
Comment (cpresser) [reply Transistor_FET.dcm:2531]: Full-text search seems to have failed me. My bad.

**Review by cpresser — CHANGES_REQUESTED** [review]
Just that one datasheet-link and we are good to go.

**Comment by cpresser** [comment]
Looks like I did not search good enough for a few of those devices.
One could argue if 5V can be considered logic-level nowadays. But I don't want to go down that rabbit-hole. If infineon says so, who am I to argue with them^^

File: Transistor_FET.dcm (hunk 5 above)
[the remark this answers is not in the document]
Comment (marecl) [reply Transistor_FET.dcm:2514]: Corrected

**Review by cpresser — APPROVED** [review]

**Pull request merged**